cloud/pkg/apis/edgeapplication/v1alpha1/type.go

```diff
@@ -0,0 +1,221 @@
+type ImageOverrider struct {
+	// Predicate filters images before applying the rule.
+	//
+	// Defaults to nil, in that case, the system will automatically detect image fields if the resource type is
+	// Pod, ReplicaSet, Deployment or StatefulSet by following rule:
+	//   - Pod: spec/containers/<N>/image
+	//   - ReplicaSet: spec/template/spec/containers/<N>/image
+	//   - Deployment: spec/template/spec/containers/<N>/image
+	//   - StatefulSet: spec/template/spec/containers/<N>/image
+	// In addition, all images will be processed if the resource object has more than one containers.
+	//
+	// If not nil, only images matches the filters will be processed.
+	// +optional
+	Predicate *ImagePredicate `json:"predicate,omitempty"`
+
+	// Component is part of image name.
+	// Basically we presume an image can be made of '[registry/]repository[:tag]'.
+	// The registry could be:
+	// - k8s.gcr.io
+	// - fictional.registry.example:10443
+	// The repository could be:
+	// - kube-apiserver
+	// - fictional/nginx
+	// The tag cloud be:
+	// - latest
+	// - v1.19.1
+	// - @sha256:dbcc1c35ac38df41fd2f5e4130b32ffdb93ebae8b3dbe638c23575912276fc9c
+	//
+	// +kubebuilder:validation:Enum=Registry;Repository;Tag
+	// +required
+	Component ImageComponent `json:"component"`
+
+	// Operator represents the operator which will apply on the image.
+	// +kubebuilder:validation:Enum=add;remove;replace
+	// +required
+	Operator OverriderOperator `json:"operator"`
+
+	// Value to be applied to image.
+	// Must not be empty when operator is 'add' or 'replace'.
+	// Defaults to empty and ignored when operator is 'remove'.
+	// +optional
+	Value string `json:"value,omitempty"`
+}
+
+// ImagePredicate describes images filter.
+type ImagePredicate struct {
+	// Path indicates the path of target field
+	// +required
+	Path string `json:"path"`
+}
+
+// ImageComponent indicates the components for image.
+type ImageComponent string
+
+// OverriderOperator is the set of operators that can be used in an overrider.
+type OverriderOperator string
+
+// These are valid overrider operators.
+const (
+	OverriderOpAdd     OverriderOperator = "add"
+	OverriderOpRemove  OverriderOperator = "remove"
+	OverriderOpReplace OverriderOperator = "replace"
+)
+
+// Manifest represents a resource to be deployed on managed node groups.
+type Manifest struct {
+	// +kubebuilder:pruning:PreserveUnknownFields
+	runtime.RawExtension `json:",inline"`
+}
+
+// EdgeAppStatus defines the observed state of EdgeApplication.
+type EdgeAppStatus struct {
+	// Conditions contain the different condition statuses for this work.
+	// Valid condition types are:
+	// 1. Applied represents workload in EdgeApplication is applied successfully on a managed node groups.
+	// 2. Progressing represents workload in EdgeApplication is being applied on a managed node groups.
+	// 3. Available represents workload in EdgeApplication exists on the managed node groups.
```

Comment:
What's the difference between `Applied` and `Available` ?

Reply:
Reference to Karmada's resource "work", EdgeAppApplied/EdgeAppAvailable are status type, EdgeAppApplied means EdgeApplication controller already split workload to nodegroups, EdgeAppAvailable means workload status is available after deploying to nodegroups.